azext_iot/monitor/property.py

```diff
@@ -7,91 +7,202 @@
+
+        issues = self._issues_handler.get_issues_with_minimum_severity(minimum_severity)
+
+        for issue in issues:
+            issue.log()
+
+    def _validate_payload_against_interfaces(self, payload: dict, name):
+        name_miss = []
+        interface_name = name.replace(PNP_INTERFACE_PREFIX, "")
+        if self._is_interface(interface_name):
+            # if the payload is an interface then iterate thru the properties under the interface
+            for property_name in payload.keys():
+                schema = self._template.get_schema(
+                    name=property_name, interface_name=interface_name
+                )
+                if not schema:
+                    name_miss.append(property_name)
+        else:
+            # if the payload is a property then process the payload as a single unit.
+            schema = self._template.get_schema(name=name, interface_name="")
+
+            if not schema:
+                name_miss.append(name)
+
+            if self._validate_duplicate_properties(name):
+                details = strings.duplicate_property_name(
+                    name, list(self._template.interfaces.keys())
+                )
+                self._add_central_issue(severity=Severity.warning, details=details)
+
+        if name_miss:
+            details = strings.invalid_field_name_mismatch_template(
+                name_miss, self._template.schema_names
             )
+            self._add_central_issue(severity=Severity.warning, details=details)
 
-            if change_d:
-                print("Changes in desired properties:")
-                print("version :", twin.desired_property.version)
-                print(change_d)
-
-            if change_r:
-                print("Changes in reported properties:")
-                print("version :", twin.reported_property.version)
-                print(change_r)
+    def _validate_duplicate_properties(self, property_name):
+        value = (
+            sum(
+                property_name in idnumber
+                for idnumber in self._template.interfaces.values()
+            )
+            > 1
+        )
+        return value
+
+    def _add_central_issue(self, severity: Severity, details: str):
+        self._issues_handler.add_central_issue(
+            severity=severity,
+            details=details,
+            message=None,
+            device_id=self._device_id,
+            template_id=self._template.id,
+        )
 
-        time.sleep(polling_interval_seconds)
+    def _is_interface(self, interface_name):
+        # Remove PNP interface prefix to get the actual interface name
+        interface_name = interface_name.replace(PNP_INTERFACE_PREFIX, "")
+        return interface_name in self._template.interfaces
 
-        prev_twin = twin
+    def _get_device_template(self):
 
+        central_device_provider = CentralDeviceProvider(self._cmd, self._app_id)
```

Comment:
I would suggest not taking a dependency on other providers within the PropertyProvider, or we risk future circular dependency issues.

We should pass down the template as part of construction (i.e. Controller level).

If we absolutely must take a dependency on the providers, I'd suggest creating the providers in the constructor, then referencing them as needed

Reply:
#important